File: Lab06/whale.pl
#!/usr/bin/perl -w


if(@ARGV == 0) {
   die "Usage: $0 'Whale name'";
   
} else {
   my $individual = 0;
   my $pods = 0;
   $whale = $ARGV[0];
   
   while($line = <STDIN>) {
      if($line =~ /^[0-9]+ $whale$/) {
         $pods++;
         $line =~ s/[^0-9]//g;
         $individual = $individual + $line;
      }
   }
   
   print "$whale observations: $pods pods, $individual individuals \n";
}

File: Lab06/whale.py
#!/usr/bin/python

import sys, re

pods = 0
individuals = 0

if len(sys.argv) == 1:
   print "Usage %s 'Whale Name'" %sys.argv[0]
   sys.exit(1)

for line in sys.stdin:
   if(re.search(r'^[0-9]+ %s' %sys.argv[1],line)):
      pods += 1;
      individuals += int(re.sub(r'[^0-9]','',line))
      #int(line.split()[0])
      #individuals += int(re.findall('\d+',line))
      #print int(re.findall('\d+',line))

print sys.argv[1], "observations:", pods, "pods,", individuals, "individuals"
